en/computer_vision/collision_prevention.md

```diff
@@ -40,6 +41,32 @@ Enabled collision prevention by [setting the following parameter](../advanced_co
 If you are using a distance sensor attached to your flight controller for collision prevention, it will need to be [attached and configured](#rangefinder) as described in the next section.
 If you are using a companion computer to provide obstacle information see [companion setup](#companion).
 
+### MPC_COL_PREV_ANG Angle Tuning {#angle_tuning}
+
+Collision prevention calculates the vehicle speed using the minimum distance from *the set of all included sensors* in the commanded direction, where the "included set" is tuned using [MPC_COL_PREV_ANG](../advanced_config/parameter_reference.md#MPC_COL_PREV_ANG).
+
+The angle affects both what objects will be detected in the direction that the vehicle is moving, and also the angle at which it can withdraw from an obstacle after stopping.
+
+Using a larger angle includes more sensors and reduces the chance of clipping obstacles.
+However a larger angle also increases the angle at which you will detect objects when you are moving away or alongside them - e.g. it is harder to travel along the obstacle surface without getting "stuck".
+Generally 45 degrees is a good compromise value.
+
+The diagrams below are used to illustrate how the setting works, where:
+- The minimum distance `MPC_COL_PREV_D = 4m`, and the sensor range is 12m.
+- The pink line shows where the user is trying to move the vehicle (this is the same for both diagrams).
+- The blue line shows the angle where received sensor data *can be used* (and what will be discarded).
+  - This is equal to double `MPC_COL_PREV_ANG` degrees centred on the commanded setpoint.
+  - On the left diagram the `MPC_COL_PREV_D` angle is 90 degrees, so all sensors on the right hand side of the vehicle are included.
+  - On the right the angle is 45 degrees, resulting in a smaller set of sensors being used (note the sensors close to the obstacle are omitted!)
+- The red arc shows the angle where sensor data *will be used*.
+  This is the subset of sensors that are both in range of an obstacle, and within the angle defined by the blue line.
+
+![MPC_COL_PREV_ANG image](../../assets/computer_vision/collision_prevention_angle.png)
+
+Because it has such a large angle the vehicle on the left includes sensors that are within the 4m minimum range; the distance used for calculation is 4m, and the commanded setpoint is ignored. 
```

Comment:
just to clarify: We do not only use the miniumum distance to calculate the velocity but all sensor data in the red arch. It is a geometric calculation of the breaking distance considering all data inside MPC_COL_PREV_ANG. But of course the core statement is correct: Because in the left case the data contains closer bins, the velocity is more limited.

Reply:
Ah, so the large angle means that we use sensor data that is within the minimum distance, but that just means it is part of the calculation. It might have a small impact because of the geometry. In this case perhaps the vehicle would still move, just slower due to those setpoints?

Assuming that is correct, I have reworded. Is this OK?

> Because it has such a large angle, the vehicle on the left includes sensor data that is within the 4m minimum range.
This data is included in the velocity calculation, and may result in the commanded setpoint being ignored (vehicle doesn't move) or obeyed at a reduced velocity. 
The vehicle on the right uses an angle that causes the collision prevention algorithm to discard the sensor data that is closest to the obstacle.
The vehicle is therefore more likely to be able to move in the commanded direction.

Reply:
Exactly! It only stops completely if there is one sensor bin indicating 4m or closer. Otherwise it will just move the slower the closer the obstacles are.

Reply:
@baumanta Actually my text doesn't quite say what you think it does. 

Yes it says that on left the range data includes a bin within 4m min range. BUT it also says that "This may result in the commanded setpoint being ignored (vehicle doesn't move) *or obeyed at a reduced velocity.*"

So my take here was that geometry is important too. I.e. you have a whole scan of data, just because one bucket is too close doesn't mean you necessarily can't move - the geometry of all the buckets affects velocity.

But now it sounds like I misinterpretted you - any buckets are in the minimum cutoff then all movement is dead. Is that correct. 

Presumably it is still true that the velocity depends on all the buckets, it is just that part of that calculation is that if any buckets are within the minimum velocity must be zero?

Reply:
So what actually happens is the following: If any bucket in the considered data range is below the specified limit, all movement is dead. The reason for this is, that because the maximum angle for COL_PREV_ANG is 90deg, we consider at most the hemisphere of buckets centered at the command. That means any execution of the command (however slow) will bring you even closer to the bucket which is already below the limit. Therefore, if one bucket is too close -> no movement. If they just contain other distances, the executed speed depends on the geometry, actual bin values etc.